src/de.rs

```diff
@@ -1,24 +1,11 @@
-use pyo3::{types::*, Bound, PyNativeType};
+use pyo3::{types::*, Bound};
 use serde::de::{self, IntoDeserializer};
 use serde::Deserialize;
 
 use crate::error::{PythonizeError, Result};
 
 /// Attempt to convert a Python object to an instance of `T`
-#[deprecated(
-    since = "0.21.0",
-    note = "will be replaced by `depythonize_bound` in a future release"
-)]
-pub fn depythonize<'de, T>(obj: &'de PyAny) -> Result<T>
-where
-    T: Deserialize<'de>,
-{
-    let mut depythonizer = Depythonizer::from_object_bound(obj.as_borrowed().to_owned());
-    T::deserialize(&mut depythonizer)
-}
-
-/// Attempt to convert a Python object to an instance of `T`
-pub fn depythonize_bound<'py, T>(obj: Bound<'py, PyAny>) -> Result<T>
+pub fn depythonize_bound<'py, T>(obj: &Bound<'py, PyAny>) -> Result<T>
```

Comment:
I'm happy to remove the original `depythonize` in the next release. Maybe instead can we please add `depythonize` back with this new `&Bound` as input, and mark `depythonize_bound` as deprecated?

The goal with the PyO3 0.21 GIL Refs removal is that in 0.23 when we've finished the migration we will rename `PyTuple::new_bound` to just `PyTuple::new` etc. in the same way.